debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -874,7 +875,7 @@ public synchronized boolean isConnected() throws SQLException {
         if (conn == null) {
             return false;
         }
-        return !conn.isClosed();
+        return !conn.isClosed() && conn.isValid(CONNECTION_VALID_CHECK_TIMEOUT_IN_SEC);
```

Comment:
Hum, hum. This seems to issue a statement, and we call `isConnected()` a lot. I'm wondering whether we should rather have a separate method `JdbcConnection::isValid()` and call that in the snapshot change event source above? @jpechane, any thoughts?